### PR TITLE
Adds support for 8.10.1

### PR DESCRIPTION
**Pull request by joshvera**
Adds support for 8.10.1 by mostly expanding base bounds.

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -1,3 +1,8 @@
 packages: . example
 
 jobs: $ncpus
+
+source-repository-package
```
Comment (joshvera) [inline cabal.project:5]: Needs a temporary source-repository-package as this `text` commit hasn't been published to Hackage.

**Comment by tclem** [comment]
Can we add 8.10 to the ci matrix as well?

**Comment by tclem** [comment]
I forget how we consume this, do we need a patch version bump too or can you pull from this sha?

**Comment by joshvera** [comment]
We pull from the SHA so this should suffice for now.

**Pull request merged**